scripts/country-us/historical-districts.py

```diff
@@ -0,0 +1,193 @@
+
+
+def download_historic_legislators():
+    historic_legislators_url = "https://raw.githubusercontent.com/unitedstates/congress-legislators/master/legislators-historical.yaml"
+    raw_historic_legislators = requests.get(historic_legislators_url).text
+    f = open(
+        os.path.join(
+            SCRIPT_DIRECTORY,
+            "../../cache/legislators-historical.yaml"),
+        'w')
+    f.write(raw_historic_legislators)
+    f.close()
+
+
+def extract_historical_districts():
+    historic_legislator_list = yaml_load(
+        os.path.join(
+            SCRIPT_DIRECTORY,
+            '../../cache/legislators-historical.yaml'))
+
+    historical_districts = set()
+    for leg in historic_legislator_list:
+        for term in leg['terms']:
+            if 'district' in term:
+                district_tuple = (term['state'].lower(), term['district'])
+                # (value of -1 specifies unknown district)
+                if (term['district'] == -1):
+                    print("Skipping district {0}".format(district_tuple))
+                    continue
+                historical_districts.add(district_tuple)
+    return historical_districts
+
+
+def extract_existing_districts():
+    congressional_district_id_file = os.path.join(
+        SCRIPT_DIRECTORY,
+        '../../identifiers/country-us/census_autogenerated/us_congressional_districts.csv')
+
+    existing_districts = set()
+    with open(congressional_district_id_file, 'rt') as csvfile:
+        reader = csv.reader(csvfile)
+        next(reader)  # skip first line
+        for row in reader:
+            existing_districts.add(parse_division_id(row[0]))
+    return existing_districts
+
+
+def extract_at_large_districts():
+    state_division_id_file = os.path.join(
+        SCRIPT_DIRECTORY,
+        '../../identifiers/country-us/states.csv')
+    territory_division_id_file = os.path.join(
+        SCRIPT_DIRECTORY,
+        '../../identifiers/country-us/us_territories.csv')
+
+    at_large_districts = set()
+    for f in [state_division_id_file, territory_division_id_file]:
+        with open(f, 'rt') as csvfile:
+            reader = csv.reader(csvfile)
+            for row in reader:
+                at_large_districts.add(parse_division_id(row[0]))
+    at_large_districts.add(('dc', 0))  # add dc; a weird exception
+    return at_large_districts
+
+
+def current_at_large_reps(existing_districts):
+    district_dict = {}
+    for district in set(filter(None, existing_districts)):
+        state = district[0]
+        if state in district_dict:
+            district_dict[state].append(district[-1])
+        else:
+            district_dict[state] = [district[-1]]
+    return [st for st in district_dict if len(district_dict[st]) == 1]
+
+
+def write_missing_districts(missing_districts):
+    new_file_path = os.path.join(
+        SCRIPT_DIRECTORY,
+        '../../identifiers/country-us/historical/unitedstates_legislators_autogenerated/historical_congressional_districts.csv')
```

Comment:
traditional way to make this PEP8 compliant is

    new_file_path = os.path.join(SCRIPT_DIRECTORY, 
                                             '../../identifiers/country-us/historical/'
                                             '/unitedstates_legislators_autogenerated/historical_congressional_districts.csv')

(this got formatted a bit wrong, but the idea is to split the string and use implicit concatenation)